src/lib/replication/repl_dev/raft_repl_dev.cpp

```diff
@@ -599,22 +615,21 @@ AsyncNotify RaftReplDev::notify_after_data_written(std::vector< repl_req_ptr_t >
     futs.reserve(rreqs->size());
 
     // Pop any entries that are already completed - from the entries list as well as from map
-    rreqs->erase(std::remove_if(
-                     rreqs->begin(), rreqs->end(),
-                     [this, &futs](repl_req_ptr_t const& rreq) {
-                         if (rreq == nullptr) { return true; }
-
-                         if (rreq->state.load() & uint32_cast(repl_req_state_t::DATA_WRITTEN)) {
-                             m_repl_key_req_map.erase(rreq->rkey); // Remove=Pop from map as well, since it is completed
-                             RD_LOG(INFO,
-                                    "Raft Channel: Data write completed and blkid mapped, removing from map: rreq=[{}]",
-                                    rreq->to_compact_string());
-                             return true; // Remove from the pending list
-                         } else {
-                             futs.emplace_back(rreq->data_written_promise.getSemiFuture());
-                             return false;
-                         }
-                     }),
+    rreqs->erase(std::remove_if(rreqs->begin(), rreqs->end(),
+                                [this, &futs](repl_req_ptr_t const& rreq) {
+                                    if ((rreq == nullptr) || (rreq->value_inlined)) { return true; }
```

Comment:
Only change is this line 620, nothing else is changed, but this resulted in format change.